### PR TITLE
Revamp events from stg

### LINKED ISSUE #97: Courses Page
image

Courses and Events should share the same Card component

### LINKED ISSUE #98: Events Page
image

Will need an external library to implement the calendar page. This should allow maintainers to maintain the events through a list of JSON data similar to previous approach. Can share the same Card component as Courses page.

### PR DESCRIPTION
**Pull request by ivanderjmw**
Screenshot 2022-01-25 at 9 27 43 PM

### CODE REVIEW

File: src/components/EventCard/index.jsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    [theme.breakpoints.down('xs')]: {
+      flexDirection: 'column',
+      alignItems: 'flex-start',
+      '& > div > div > h6': {
+        marginTop: '0.3em',
+      },
+    },
+  },
+  dateBox: {
+    backgroundColor: '#FAE9D0',
```
Comment (jerrenss) [inline src/components/EventCard/index.jsx]: For all CSS color hex codes, can we reuse them from the palette in /theme folder? If the hex code is not there, can add them in too.

File: src/components/EventCard/index.jsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+const EventCard = ({ date, location, title, description, imageUrl }) => {
+  const classes = useStyles()
+
+  return (
+    <>
+      <Box className={classes.root}>
+        <Box
+          className={classes.imageCrop}
+          style={{ backgroundImage: `url(${imageUrl || 'default-event.jpg'})` }}
```
Comment (jerrenss) [inline src/components/EventCard/index.jsx:130]: For the placeholder image, which i think most of the time will be used, can we optimise/find a smaller sized image? Currently 1.79MB quite large for an image hahah which might not load instantly

File: src/components/EventCard/index.jsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+            <br />
+            {moment(date).format('D')}
+          </Box>
+          <Box className={classes.dateBoxXS}>
+            {moment(date).format('MMMM D')}
+          </Box>
+          <Box>
+            <Box className={classes.location}>
+              <LocationOn />
+              <Typography variant="body1">{location}</Typography>
```
Comment (jerrenss) [inline src/components/EventCard/index.jsx]: Typography default is body1 so can omit

File: src/views/Events/index.jsx [hunk 4]
```diff
@@ -0,0 +1,207 @@
+import { Box, Container, makeStyles, Typography } from '@material-ui/core'
+import moment from 'moment'
+import { Calendar, momentLocalizer } from 'react-big-calendar'
+import EventCard from '../../components/EventCard'
+import EventSpotlight from '../../components/EventSpotlight'
+import Layout from '../../components/Layout'
+
+import 'react-big-calendar/lib/css/react-big-calendar.css'
+// import './react-big-calendar.css'
```
Comment (jerrenss) [inline src/views/Events/index.jsx]: Can remove i guess

File: src/views/Events/index.jsx [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+
+const useStyles = makeStyles((theme) => ({
+  root: {},
```
Comment (jerrenss) [inline src/views/Events/index.jsx]: Can remove also if there is no styling for the root component

File: src/views/Events/index.jsx [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+        display: 'flex',
+        justifyContent: 'center',
+      },
+    },
+  },
+}))
+
+const localizer = momentLocalizer(moment)
+
+const events = [
```
Comment (jerrenss) [inline src/views/Events/index.jsx]: Oh for this, can we place it in a src/data/events.js file, so it is easy to maintain the records all in one place

File: src/views/Events/index.jsx [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+    end: new Date(),
+  },
+  {
+    id: 2,
+    title: 'It will be there yeah',
+    location: 'NUS School of Computing',
+    date: new Date('01/27/2022'),
+    description:
+      ':S DjJ ;lakJD;flkds;lfja;mx clkvnsghsdfja;dlskjfa;dlkj fa;lkjd ;alknvlcnva;lsdn fasdf;laskjd f;alkmvlxcnv;',
+    allDay: true,
```
Comment (jerrenss) [inline src/views/Events/index.jsx]: Also, can we have a standard schema for events? I noticed some records have more fields than others. Eg. allDay can be true or false for different records.

File: src/views/Events/index.jsx [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+                defaultDate={new Date()}
+                localizer={localizer}
+                popup={true}
+                startAccessor="start"
+                endAccessor="end"
+                titleAccessor="title"
+              />
+            )}
+          </Box>
+          <Box className="event">
```
Comment (jerrenss) [inline src/views/Events/index.jsx:112]: Oh question, do you intend to include logic to dynamically display the spotlight and various events? Seems like now the spotlight is the first in our event records and and the various events follow.

I was thinking maybe can consider:
- For each day / month / week view, we only show events that are during that time frame. The spotlight will default be the earliest upcoming event
- Alternatively, we can show only events that have not been completed. The spotlight will default be the earliest upcoming event

**Review by jerrenss — CHANGES_REQUESTED** [review]
Thanks for the great work! Just noted some improvements which can be made :)

File: src/views/Events/index.jsx (hunk 8 above)
Comment (ivanderjmw) [reply src/views/Events/index.jsx:112]: I think the second one will be better. The spotlight is the earliest event.

**Comment by jerrenss** [comment]
Looks good, just need resolve conflicts and place the hex codes in theme! @ivanderjmw

**Comment by ivanderjmw** [comment]
@jerrenss this includes events & courses, please check :)

**Review by jerrenss — APPROVED** [review]
Thanks for the good work, will create another issue for minor improvements as a separate task!

**Pull request merged**